### PR TITLE
Optimize Dockerfile

### PR DESCRIPTION
**Pull request by MilesChou**
Adjust build order for [leverage build cache suggestion](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,20 @@
 FROM golang:1.13-alpine3.11 AS builder
 
+RUN apk add -U --no-cache \
+        ca-certificates
+
+WORKDIR /go/src/github.com/minchao/go-realworld
+COPY go.* ./
```
Comment (minchao) [inline Dockerfile]: How about `COPY go.mod go.sum ./`?

Comment (MilesChou) [reply Dockerfile]: Got it, done.

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+# build application
 ARG BUILD_VERSION
 ARG BUILD_DATE
 ARG BUILD_COMMIT
 ARG CMD_PACKAGE
 
-WORKDIR /go/src/github.com/minchao/go-realworld
 COPY . .
-RUN apk add -U --no-cache \
-        ca-certificates
```
Comment (minchao) [inline Dockerfile:11]: Always get latest version of ca-certificates.
Maybe we can add version to package like `ca-certificates=20191127-r0`, and updated regularly?

Comment (minchao) [reply Dockerfile:11]: update with base image

**Pull request merged**